language/move-prover/bytecode/src/read_write_set_analysis.rs

```diff
@@ -879,3 +842,67 @@ impl Default for ReadWriteSetState {
         }
     }
 }
+
+// =================================================================================================
+// Converting Infer Result into standalone types
+
+impl Offset {
+    pub fn make_canonical(&self, env: &GlobalEnv) -> Option<RWOffset> {
+        Some(match self {
+            Offset::Field(idx) => RWOffset::Field(*idx),
+            Offset::VectorIndex => RWOffset::VectorIndex,
+            Offset::Global(s) => RWOffset::Global(s.get_type().into_struct_type(env)?),
+        })
+    }
+}
+impl AccessPath {
+    pub fn make_canonical(&self, env: &GlobalEnv) -> Option<Vec<RWAccessPath>> {
+        let (roots, access_opt) = match self.root() {
+            Root::Formal(idx) => (vec![RWRoot::Formal(*idx)], None),
+            Root::Global(key) => (
+                key.address()
+                    .get_concrete_addresses()
```

Comment:
I think we need to change `get_concrete_addresses()` to return `Option<Vec<AccountAddress>>` + be careful to return `None` here if that function returns `None`.

Reply:
I'll fix this in the final PR.